sdks/python/apache_beam/runners/portability/fn_api_runner/fn_runner_test.py

```diff
@@ -510,6 +510,21 @@ def process(
       actual = (p | beam.Create(data) | beam.ParDo(ExpandingStringsDoFn()))
       assert_that(actual, equal_to(list(''.join(data))))
 
+  def test_sdf_with_dofn_as_restriction_provider(self):
```

Comment:
Can we add WatermarkEstimatorProvider for testing as well?

Reply:
Done.